Fix doc EXAMPLES formatting for HTML

mislav opened the pull request:
Example contents like below ~~should be intended per Markdown syntax for code blocks~~ so they don't get rendered as prose (all on the same line) in our web docs:
```
$ gh issue list
$ gh pr create
```

Also, there were some incorrect docs.

Followup to #1106
Ref. #1166

mislav committed:
Fix EXAMPLES sections of various commands

- Code should be intented per Markdown syntax for a code block. This
  matters when this documentation is rendered to HTML

- Fix some innaccurate usage examples

- Tweak wording, formatting in a few places

mislav committed:
Fix FLAGS section showing up empty for `config get/set`

This is because `config get/set` both do have a flag, but it's hidden.

mislav commented on command/alias.go:
```diff
@@ -23,24 +23,30 @@ var aliasCmd = &cobra.Command{
-	Long:               `This command lets you write your own shortcuts for running gh. They can be simple strings or accept placeholder arguments.`,
-	Example: `
-	gh alias set pv 'pr view'
-	# gh pv -w 123 -> gh pr view -w 123.
```
I found this hard to parse, so I've broken it down in multiple lines

mislav commented on command/alias.go:
```diff
@@ -23,24 +23,30 @@ var aliasCmd = &cobra.Command{
+	Long: `Declare a word as a command alias that will expand to the specified command.
 
-	gh alias set epicsBy 'issue list --author="$1" --label="epic"'
-	# gh epicsBy vilmibm -> gh issue list --author="$1" --label="epic"
```
`$1` wasn't expanded

mislav commented on command/alias.go:
```diff
@@ -168,11 +174,10 @@ func aliasList(cmd *cobra.Command, args []string) error {
-	Use:     "delete <alias>",
-	Short:   "Delete an alias.",
-	Args:    cobra.ExactArgs(1),
-	Example: "gh alias delete co",
```
I didn't believe this example added anything over the usage line already displayed: `gh alias delete <alias>`, so I've removed it

mislav commented on command/credits.go:
```diff
@@ -44,10 +44,15 @@ var creditsCmd = &cobra.Command{
 	Short: "View credits for this tool",
 	Long:  `View animated credits for gh, the tool you are currently using :)`,
-	Example: `gh credits            # see a credits animation for this project
-  gh credits owner/repo # see a credits animation for owner/repo
```
This version of `credits` does not support the repo argument

mislav commented on command/gist.go:
```diff
@@ -27,20 +27,29 @@ var gistCmd = &cobra.Command{
 }
 
 var gistCreateCmd = &cobra.Command{
-	Use:   `create {<filename>|-}...`,
```
`-` cannot be repeated multiple times, therefore a more precise syntax would be `{<filename>...|-}`. Furthermore, the argument is optional in piped mode, so `[<filename>... | -]`

mislav commented on command/gist.go:
```diff
@@ -27,20 +27,29 @@ var gistCmd = &cobra.Command{
 	Short: "Create a new gist",
-	Long: `gh gist create: create gists
-
-Gists can be created from one or many files. This command can also read from STDIN. By default, gists are private; use --public to change this.
```
I have tried to wrap any long lines

mislav commented on command/issue.go:
```diff
@@ -52,9 +52,11 @@ var issueCmd = &cobra.Command{
 	Short: "Create and view issues",
 	Long:  `Work with GitHub issues`,
-	Example: `$ gh issue list
-$ gh issue create --fill
```
There is no `issue create --fill`

mislav commented on command/pr.go:
```diff
@@ -63,9 +64,11 @@ $ gh pr view --web`,
 var prListCmd = &cobra.Command{
 	Use:   "list",
 	Short: "List and filter pull requests in this repository",
-	Example: `$ gh pr list --limit all
```
There is no `--limit all`

mislav commented on command/pr.go:
```diff
@@ -63,9 +64,11 @@ $ gh pr view --web`,
 	Short: "List and filter pull requests in this repository",
-	Example: `$ gh pr list --limit all
-$ gh pr list --state closed
-$ gh pr list --label “priority 1” “bug”`,
```
1. This was incorrect syntax for multiple labels
2. Curly quotes are not supported in shells, so I've replaced them with normal double quotes

probablycorey replied:
This is what I get for copy/pasting directly from a google doc 😞

mislav commented on command/pr_review.go:
```diff
@@ -23,18 +23,24 @@ func init() {
-
-Examples:
-
-	gh pr review                                  # add a review for the current branch's pull request
```
Something bothered me about the `branch's` contraction, so I've avoided it by rewording

mislav commented on command/root.go:
```diff
@@ -96,9 +96,11 @@ var RootCmd = &cobra.Command{
 	SilenceErrors: true,
 	SilenceUsage:  true,
-	Example: `$ gh issue create
-$ gh repo clone
```
`repo clone` without arguments is not a valid invocation

mislav commented on command/repo.go:
```diff
@@ -75,15 +76,17 @@ To pass 'git clone' flags, separate them with '--'.`,
 	Use:   "create [<name>]",
 	Short: "Create a new repository",
-	Long:  `Create a new GitHub repository`,
-	Example: utils.Bold("$ gh repo create") + `
```
We should not use ANSI escape sequences for terminal colors here because this text is also used to generate man pages and HTML pages, where those ANSI sequences are not supported.

probablycorey reviewed (approved):
Thanks for this cleanup @mislav 🥇. I didn't realize the examples section was used to populate our HTML docs. I was only focused on the terminal help output.

mislav committed:
Fix indentation of Example blocks

In HTML, `Example` blocks seem to be already injected in fenced Markdown
blocks `` ``` ``, so they don't need to be especially intented.

mislav committed:
Merge remote-tracking branch 'origin' into doc-examples-tweaks

mislav committed:
Tweak last few Example sections

mislav commented:
Pushed some more fixes! I've realized that Example blocks don't really need to be indented to become code blocks in HTML because our HTML renderer already puts them in `` ``` ``, so I've wrapped them in handy `heredoc.Doc()` helper that allows heredoc-style string literals in go code, but processes them so they are de-dented and their extra line breaks trimmed.
